src/hardware/serialport/softmodem.cpp

```diff
@@ -325,6 +325,7 @@ void CSerialModem::AcceptIncomingCall() {
 	if (waitingclientsocket) {
 		clientsocket = std::move(waitingclientsocket);
 		EnterConnectedState();
+		warmuptimer = MODEM_CALLWARMUPTIME;
```

Comment:
I can't see where `MODEM_CALLWARMUPTIME` is defined (or as a `constexpr`)?

Reply:
Despite this code using smashedlowercasenames, suggest using `snake_case` variable and local static functions to match the Staging additions.

MODEM_CALLWARMUPTIME:

This `TIME` is used as an absolute here, like: `DINNERTIME`. My understanding is that this is a count-down? In milliseconds?

Suggest something like: `MODEM_WARMUP_MS`.

And then the variable indicates how many remaining milliseconds are left in the warmup.

```warmuptimer -> remaining_warmup_ms```

Remaining then naturally fits with decrementing it, like: `remaining_warmup_ms--`.

Reply:
`remaining_warmup_ms` is too long, need something more concise.

Reply:
> `remaining_warmup_ms` is too long

There are many variables longer than this because they made most sense. 

If you can describe it more concisely, please do, however, dosbox staging the project has never constrained variable length just for the sake of length alone. 


> `_ms` suffix also makes no sense since it's actually number of modem ticks and a single tick may not equal 1 ms.

There are only two emulated ticks in DOSBox: 1ms of emulated time (which yes, in host time might not be perfectly 1ms-- however, the DOS world experiences exactly 1ms of time). 

And there are  BIOS clock ticks, which are made up of ~18 ms worth of DOS time (derived from the emulated 1ms). 

Which every tick countdown this represents, let's describe it as such.  `remaining_warmup_dosbox_ticks`, if you'd rather that. 
Or `remaining_warmup_bios_click_ticks`, if they're multiplied up from the BIOS clock.

Reply:
> There are only two emulated ticks in DOSBox: 1ms of emulated time (which yes, in host time might not be perfectly 1ms-- however, the DOS world experiences exactly 1ms of time).

Clearly, you've forgotten our earlier discussion about modem tick rate. This snippet right here defines modem tick rate, i.e. the frequency at which `Timer2()` is called. It is currently set to 1 ms (i.e. 1000 ticks per second).
https://github.com/dosbox-staging/dosbox-staging/blob/54d5830c95ef0e55fa700de1ff6b40cc2195022f/src/hardware/serialport/softmodem.cpp#L211-L219

Reply:
As for var name, it was named in accordance with the other two variables used for countdown - `ringtimer` and `dtrofftimer`

Reply:
```
src/hardware/serialport/softmodem.h:#define MODEM_TICKRATE 1000 // Ticks per second
src/hardware/serialport/softmodem.h:#define MODEM_TICKTIME (1000 / MODEM_TICKRATE) // Tick interval in milliseconds
```

Looks like `MODEM_TICKRATE` is 1ms of DOS time.

> in accordance with the other two existing variables used for countdowns - ringtimer and dtrofftimer

When adding new code, we'd rather it conform to our own standards: and that's `snake_case` for variables involving time, the variable should include some description of the purpose to help other developers clearly understand it:
 - Is it a "rate", which mean events-per-s?
 - Is it a "period", which means time-per-event?
 - Is it a "tally" (incrementing) or "countdown" (decrementing)
 - What units are the time in? milliseconds / ticks/ etc.

Some recent examples of DOSBox Staging additions of names:

 - `TITLEBAR_CYCLES_MS`

```
src/hardware/compressor.cpp:    assert(attack_time_ms > 0.0f);
src/hardware/compressor.cpp:    assert(release_time_ms > 0.0f);
src/hardware/compressor.cpp:    assert(rms_window_ms > 0.0f);
src/hardware/compressor.cpp:    attack_coeff    = expf(-1.0f / (attack_time_ms * sample_rate_hz));
src/hardware/compressor.cpp:    release_coeff   = expf(-millis_in_second_f / (release_time_ms * sample_rate_hz));
src/hardware/compressor.cpp:    rms_coeff       = expf(-millis_in_second_f / (rms_window_ms * sample_rate_hz));

```

I suggest a name just as an example. Please feel free to use one that you think is best.

Reply:
Updated PR, how's this?